### PR TITLE
DevshellPackages: Update to python3.11

### PR DESCRIPTION
**Pull request by mitzkia**
Signed-off-by: Andras Mitzki <andras.mitzki@quest.com>

This PR tries to fix the issue related to building devshell image

### CODE REVIEW

**Comment by alltilla** [comment]
Can you please do a test run with the updated devshell image? Thanks!

**Comment by mitzkia** [comment]
Of course, sure.

File: dbld/packages.manifest [hunk 1]
```diff
@@ -96,8 +96,9 @@ python3-pip                     [debian, ubuntu]
 
 # python 3.10 has issues on Debian Testing
 # Nevertheless we should only downgrade for Light style-check tests, which uses pre-commit
-python3.9                   [devshell]
-python3.9-venv              [devshell]
+python3.11                   [devshell]
+python3.11-dev               [devshell]
```
Comment (gaborznagy) [inline dbld/packages.manifest:100]: IMHO the `-dev` package is not needed. This python installation for `devshell` is added so the Light style-check (using pre-commit) works, it's not for syslog-ng's python dependency.
python3 is installed based on `packaging/debian/control` and `dbld/build.manifest` files.

Installation of this specific python-version was added in e61d4cf2caa49d108f28bfa884763d1bbb4d990b when we found a pre-commit issue.

Comment (mitzkia) [reply dbld/packages.manifest:100]: Thanks for the review. Unfortunately `python3.11-dev` package will be required (if possible I would skip it also), due to `psutil` package will not be installed (built) without it during `pip install`. Yes, I know previously it was not required, but in `python3.11` it needed.

**Comment by mitzkia** [comment]
@gaborznagy Thanks for the review comments, I will squash the commits.

**Comment by mitzkia** [comment]
I have run GitHub Actions jobs for the updated devshell image.
Here is the building of new devshell image into my fork (uploaded image id: 10ce80da):
https://github.com/mitzkia/syslog-ng/runs/7948340687?check_suite_focus=true

Here are the GithubActions CI runs (separate branch has been created for using proper devshell images from my fork):
https://github.com/mitzkia/syslog-ng/runs/7948535758?check_suite_focus=true

This is the raw output, where you can see the proper image (10ce80da) has been downloaded:
https://pipelines.actions.githubusercontent.com/serviceHosts/25ec8f32-2f66-4c50-8b65-b5e82d1abe70/_apis/pipelines/1/runs/3360/signedlogcontent/6?urlExpires=2022-08-22T09%3A59%3A35.4469786Z&urlSigningMethod=HMACV1&urlSignature=11KndbG%2FsPwFnoOx%2BAUxx9FOW2lBvsz4NefNZtnY0PQ%3D

I will rebase this PR to the master, to see the last unit-test failure (related to patterndb) has been fixed.

**Review by MrAnno — APPROVED** [review]

File: dbld/packages.manifest (hunk 1 above)
Comment (MrAnno) [reply dbld/packages.manifest:100]: (psutil is currently compiled from source with the current state of pypi and debian-testing)

**Review by gaborznagy — APPROVED** [review]
Thanks @mitzkia !

**Pull request merged**